api/v1alpha1/remoteistio_types.go

```diff
@@ -0,0 +1,258 @@
+// the RemoteIstio CR is updated.
+type RemoteIstioUpdateStrategy struct {
+	// Type of strategy to use. Can be "InPlace" or "RevisionBased". When the "InPlace" strategy
+	// is used, the existing Istio control plane is updated in-place. The workloads therefore
+	// don't need to be moved from one control plane instance to another. When the "RevisionBased"
+	// strategy is used, a new Istio control plane instance is created for every change to the
+	// RemoteIstio.spec.version field. The old control plane remains in place until all workloads have
+	// been moved to the new control plane instance.
+	//
+	// The "InPlace" strategy is the default.	TODO: change default to "RevisionBased"
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=1,displayName="Type",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:select:InPlace", "urn:alm:descriptor:com.tectonic.ui:select:RevisionBased"}
+	// +kubebuilder:validation:Enum=InPlace;RevisionBased
+	// +kubebuilder:default=InPlace
+	Type UpdateStrategyType `json:"type,omitempty"`
+
+	// Defines how many seconds the operator should wait before removing a non-active revision after all
+	// the workloads have stopped using it. You may want to set this value on the order of minutes.
+	// The minimum and the default value is 30.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=2,displayName="Inactive Revision Deletion Grace Period (seconds)",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:number"}
+	// +kubebuilder:validation:Minimum=30
+	InactiveRevisionDeletionGracePeriodSeconds *int64 `json:"inactiveRevisionDeletionGracePeriodSeconds,omitempty"`
+
+	// Defines whether the workloads should be moved from one control plane instance to another
+	// automatically. If updateWorkloads is true, the operator moves the workloads from the old
+	// control plane instance to the new one after the new control plane is ready.
+	// If updateWorkloads is false, the user must move the workloads manually by updating the
+	// istio.io/rev labels on the namespace and/or the pods.
+	// Defaults to false.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=3,displayName="Update Workloads Automatically",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	UpdateWorkloads bool `json:"updateWorkloads,omitempty"`
+}
+
+// RemoteIstioStatus defines the observed state of RemoteIstio
+type RemoteIstioStatus struct {
+	// ObservedGeneration is the most recent generation observed for this
+	// RemoteIstio object. It corresponds to the object's generation, which is
+	// updated on mutation by the API Server. The information in the status
+	// pertains to this particular generation of the object.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Represents the latest available observations of the object's current state.
+	Conditions []RemoteIstioCondition `json:"conditions,omitempty"`
```

Comment:
The only reason we're using our own `RemoteIstioCondition` is so that we can use a custom type (enum) for `condition.type` and `condition.reason`. But now that you mention this, I wonder how useful it is to have these custom types. The downside of having a custom `RemoteIstioCondition` is that we're missing a bunch of validation tags that were added to `metav1.Condition` after we created the custom type.